### DISCUSSION TITLE
About BasicPublish.How do I know whether the transmission is successful?

### DISCUSSION THREAD

**Discussion by a645010545** [rabbitmq/rabbitmq-dotnet-client#1351] [General]
### Is your feature request related to a problem? Please describe.

When a message is sent to the queue, how do I determine whether the message is sent successfully?

### Describe the solution you'd like

BasicPublish adds a return value and can resend a post if it fails.

### Describe alternatives you've considered

Now there is no return value, we do not know whether the transmission was successful.

**Comment by michaelklishin** [comment]
Consider using GitHub Discussions for questions. They are no longer a new and relatively unknown GitHub features.

**Comment by michaelklishin** [comment]
We will not be adding a return value and will not be making `basic.publish` a synchronous method because [publisher confirms](https://rabbitmq.com/confirms.html) have been around for more than a decade. See [Publisher-side Data Safety](https://rabbitmq.com/publishers.html#data-safety).

**Comment by lukebakken** [comment]
@a645010545 please see this tutorial - https://www.rabbitmq.com/tutorials/tutorial-seven-dotnet.html
